test/staging/RandomNumberConsumer.spec.js

```diff
@@ -26,37 +28,49 @@ developmentChains.includes(network.name)
-                      fundAmount,
-                      ethers.utils.defaultAbiCoder.encode(["uint64"], [subscriptionId])
-                  )
+                  console.log(`subscription created successfully, and subscription Id is ${subscriptionId}`)
               }
 
-              const randomNumberConsumerV2Factory = await ethers.getContractFactory(
-                  "RandomNumberConsumerV2"
+                const fundAmount = networkConfig[chainId]["fundAmount"]
+                const linkTokenAddress = networkConfig[chainId]["linkToken"]
+                const linkToken = new ethers.Contract(linkTokenAddress, LINK_TOKEN_ABI, deployer)
+                console.log(`Transfer ${fundAmount} LINK to subscription ${subscriptionId}..... `)
+
+                await linkToken.transferAndCall(
+                    vrfCoordinatorAddress,
+                    fundAmount,
+                    ethers.utils.defaultAbiCoder.encode(["uint256"], [subscriptionId])
+                )
+              
+
+              const randomNumberConsumerV2PlusFactory = await ethers.getContractFactory(
+                  "RandomNumberConsumerV2Plus"
               )
-              randomNumberConsumerV2 = await randomNumberConsumerV2Factory
+              console.log("Deploying the VRF consumer smart contract......")
+
+              randomNumberConsumerV2Plus = await randomNumberConsumerV2PlusFactory
                   .connect(deployer)
                   .deploy(subscriptionId, vrfCoordinatorAddress, keyHash)
+              console.log(`VRF consumer deployed successfully at ${randomNumberConsumerV2Plus.address}`)
 
-              await vrfCoordinator.addConsumer(subscriptionId, randomNumberConsumerV2.address)
+              console.log(`Adding consumer ${randomNumberConsumerV2Plus.address} to coordinator`)
+              await vrfCoordinator.addConsumer(subscriptionId, randomNumberConsumerV2Plus.address)
           })
 
           it("Our event should successfully fire event on callback", async function () {
               // we setup a promise so we can wait for our callback from the `once` function
               await new Promise(async (resolve, reject) => {
-                  // setup listener for our event
-                  randomNumberConsumerV2.once("ReturnedRandomness", async () => {
+                // setup listener for our event
+                  randomNumberConsumerV2Plus.once("ReturnedRandomness", async () => {
                       console.log("ReturnedRandomness event fired!")
-                      const firstRandomNumber = await randomNumberConsumerV2.s_randomWords(0)
-                      const secondRandomNumber = await randomNumberConsumerV2.s_randomWords(1)
+                      const firstRandomNumber = await randomNumberConsumerV2Plus.s_randomWords(0)
+                      console.log(`first random number is ${firstRandomNumber}`)
+                      const secondRandomNumber = await randomNumberConsumerV2Plus.s_randomWords(1)
```

Comment:
Nit: should we remove the console.logs from tests?

Also -- should we not update to ethers v6? the .gt() commands etc will change. V6 uses BigInt which handles simple comparison operators like > < etc

Reply:
Yes, I will remove the cosole.log from test. These logs are for debugging when I upgrade the VRF version. 

For updates from ether v5 to v6, it will influence our existing unit and staging tests greatly. Not only VRF, but all tests including data feed, and functions will not work.  If we want to upgrade ethers.js version. Maybe we can create a new PR.

Reply:
agreed - lets do in a separate PR.